src/main/java/fr/openent/competences/service/impl/DefaultExportService.java

```diff
@@ -2465,7 +2465,7 @@ private void handleHasDevoirMatiere(JsonObject matiere, List<Service> services,
                 }
                 if (currentCoeff != 1)
                     sousMatiere.put(Field.COEFF, currentCoeff);
-                if (moySousMatiere != null) {
+                if (moySousMatiere != null && !Field.NN.equals(moySousMatiere.getValue(Field.UNROUND_AVERAGE).toString())) {
```

Comment:
Peut-être le simplifié en passant directement la méthode getString ?

```suggestion
              if (moySousMatiere != null && !Field.NN.equals(moySousMatiere.getString(Field.UNROUND_AVERAGE))) {
```

Reply:
Je n'ai pas utilisé getSring car cela peut-être aussi un double

Reply:
c'est vrai ! j'avais oublié cette histoire aussi 😆

Reply:
Attention si .getValue(Field.UNROUND_AVERAGE) renvois un double alors 
`moySousMatiere.getValue(Field.UNROUND_AVERAGE).toString()` fonctionne mais `moySousMatiere.getString(Field.UNROUND_AVERAGE)` levera un CastException.

Il faut être sur que getValue(Field.UNROUND_AVERAGE) est tout le temps un string

```
    public String getString(String key, String def) {
        Objects.requireNonNull(key);
        CharSequence cs = (CharSequence)this.map.get(key); ❌
        return cs == null && !this.map.containsKey(key) ? def : (cs == null ? null : cs.toString());
    }
```

Pour moi il ne faut pas changer.

Reply:
ouai, on a écrit pareil en plus rapide 😛